src/components/Controls/__tests__/Controls.test.js

```diff
@@ -21,35 +21,58 @@ describe("Controls", () => {
     expect(wrapper.find(`[label="Pot"]`)).toHaveLength(1);
     expect(wrapper.find(`[label="Max"]`)).toHaveLength(1);
     expect(wrapper.find(`[label="Fold"]`)).toHaveLength(1);
-    expect(wrapper.find(`[label="Call"]`)).toHaveLength(1);
+    expect(wrapper.find(`[label="Check"]`)).toHaveLength(1);
     expect(wrapper.find(`[label="Raise to"]`)).toHaveLength(1);
     expect(wrapper.find("Slider")).toHaveLength(1);
   });
 
-  test("shows the Call amount on the Call button when raise is not All-In", () => {
+  test("shows the Call button and the call amount", () => {
     const state = testState;
     state.toCall = 100;
+    state.controls.canCheck = false;
     const wrapper = buildWrapper(state);
 
     expect(wrapper.find(`[label="Call"]`)).toHaveLength(1);
     expect(wrapper.find(`[label="Call"]`).text()).toBe("Call 100");
   });
 
-  test("shows the Raise amount on the Raise button when raise is not All-In", () => {
+  test("shows the Raise button with the amount when Raise is not All-In", () => {
     const state = testState;
     state.minRaiseTo = 100;
+    state.controls.canRaise = true;
     const wrapper = buildWrapper(state);
 
     expect(wrapper.find(`[label="Raise to"]`)).toHaveLength(1);
     expect(wrapper.find(`[label="Raise to"]`).text()).toBe("Raise to 100");
   });
 
+  test("shows the Check button instead of the Call button", () => {
+    const state = testState;
+    state.toCall = 100;
+    state.controls.canCheck = true;
+    const wrapper = buildWrapper(state);
+
+    expect(wrapper.find(`[label="Call"]`)).toHaveLength(0);
```

Comment:
Thanks! The state has lots of properties, and since I am using `mount` instead of `shallow`, the test will need to access many of the properties. That's why I decided to create a separate `testState` file, to separate the actual state from the test one.

This might not be the best solution, as you have pointed out. Would defining all of the data attributes in the test file be a better solution? I get that it would be easier to see the context, but I feel like it would complicate things, especially if the structure of the state object would change.

Reply:
Oh, now I get it, thank you. I've made changes in 6b92dff to address this.